src/shop-detail.html

```diff
@@ -283,6 +325,10 @@
         if (!offline) {
           this._tryReconnect();
         }
+      },
+
+      _hasPaymentRequest: function() {
```

Comment:
it could be removed since the contract would just be `this.$.payment.request()` and the error/success events.